File: testsuite/matlab_and_octave/kernel.m
function y = kernel(filename)
	addpath('util');
	addpath('../data/kernel');
	y=true;

	eval('globals'); % ugly hack to have vars from filename as globals
	system(sprintf('ln -sf ../data/kernel/%s.m testscript.m', filename));
	testscript;
	system('rm -f testscript.m'); %avoid ultra long filenames (>63 chars)
	%eval(filename);

	if ~set_features()
		return;
	end

	if ~set_kernel()
		return;
	end

	kmatrix=sg('get_kernel_matrix');
	ktrain=max(max(abs(km_train-kmatrix)));

	sg('init_kernel', 'TEST');
	kmatrix=sg('get_kernel_matrix');
	ktest=max(max(abs(km_test-kmatrix)));

	data={'kernel', ktrain, ktest};
	y=check_accuracy(accuracy, data);

File: testsuite/r-modular/kernel.R
kernel <- function() {
	source('util/get_features.R')
	source('util/get_kernel.R')
	source('util/check_accuracy.R')

	feats <- get_features('kernel_')
	if (typeof(feats)=='logical') {
		return(TRUE)
	}

	kernel <- get_kernel(feats)
	if (typeof(kernel)=='logical') {
		return(TRUE)
	}

	kmatrix <- kernel$get_kernel_matrix()
	km_train <- max(max(abs(kernel_matrix_train-kmatrix)))

	kernel$init(kernel, feats[[1]], feats[[2]])
	kmatrix <- kernel$get_kernel_matrix()
	km_test <- max(max(abs(kernel_matrix_test-kmatrix)))
	data <- list(km_train, km_test)
	return(check_accuracy(kernel_accuracy, 'kernel', data))
}
